### ISSUE TITLE
caire never completes

### ISSUE THREAD

**Issue by foobargeez** [esimov/caire#1]
Hello,
I just gave caire a try against one of my images and it seems to be spinning non-stop.

`./caire -in ./source.JPG -out ./dest.JPG -width 480 -height 720`

And, I then changed to `perc 20` and this worked but produced the image that is almost 3 times the original size.

Not sure what I am missing.  Also, `-debug` doesn't print anything ...

Let me know what additional info you need ...

Thanks!

**Comment by esimov** [comment]
In normal circumstances the command should look like something like this:

`caire -in input/source.jpg -out ./out.jpg -width 480 -height 0 -debug true`

Please take note the process in case the image is too big and you want to reduce to a considerable size, will take quite a long time. In case you use percentage with `-perc` flag you should only provide `true` or `false` as this is a boolean flag. Then you can provide the desired size reduction in terms of percentage. 

Let me know if it's working.

**Comment by foobargeez** [comment]
Thanks.

Regarding `-perc`, can you please provide an example?  I tried as `-perc true 20` and no matter what the percentage value is, the image is always 3-4 times the original size.

I tried your command for width/height and you're right, it's taking considerable time -- it does complete though.  I deal with tens of thousands of images on a daily basis, so at the moment, this is too slow for me.  I will be watching the repo though.  Good luck!

**Comment by esimov** [comment]
Regarding `-perc` as i said it's a boolean flag, which means you need only to specify that you want to deal with percentages instead of concrete values.  Here is a sample command using `-perc`:

`caire -in input/source.jpg -out ./out.jpg -perc true -width 20 -height 20 -debug false`

which reduces the image width & height 20%.

**Comment by foobargeez** [comment]
I just tried that -- my output image size only increases it's size by 3-4 times no matter what values I use and the image width and height doesn't change.

Thanks in advance!

**Comment by esimov** [comment]
Sorry, try it with `-perc=1`.

**Comment by foobargeez** [comment]
Yep, that worked (the output image size is still 3 times more) and in my case, the image lost its visual appeal -- see attached:

Orig:
![input](https://user-images.githubusercontent.com/8659389/35529918-7fd0edfa-0500-11e8-8084-57594bd66332.jpg)

Modified:
![new](https://user-images.githubusercontent.com/8659389/35529924-8311370e-0500-11e8-9ac3-9fbd9dda9b79.jpg)

**Comment by esimov** [comment]
Be aware, this method is not suitable for any kind of images. One kind of image the algorithm is not performing quite well is when there are many important parts in it and you want to reduce the image below the accepted size. For this reason i have included the option to modify the blur radius and sobel filter threshold, but this does not guarantee that the output will be without artifacts. 

I will include in the next step a face recognition algorithm, to reduce the possible deformation on face part.

**Comment by foobargeez** [comment]
> this method is not suitable for any kind of images

That's the hard part and the challenge is equally applicable for both of us.  It's nearly impossible for me to segregate images based on what the tool can do well or otherwise, especially when I am dealing with hundreds of thousands of images.

I will be subscribing to future updates on this project.  Good luck!
